client/components/artwork/ArtworkList.js

```diff
@@ -0,0 +1,129 @@
+import React, { PropTypes } from 'react';
+import { GridList } from 'material-ui/GridList';
+import Subheader from 'material-ui/Subheader';
+import RaisedButton from 'material-ui/RaisedButton';
+import ArtworkListItem from './ArtworkListItem';
+
+/**
+ * Stateless Component responsible for rendering the a list of Artworks by Categories.
+ * This categories are movieId and languageCode.
+ * PAGE_SIZE controls how many sections will be shown per offset.
+ */
+
+const PAGE_SIZE = 1;
+const styles = {
+  container: {
+    display: 'flex',
+    marginTop: '60px',
+    flexWrap: 'wrap',
+    justifyContent: 'space-around'
+  },
+  gridList: {
+    width: '100%',
+    height: '100%',
+    overflowY: 'auto'
+  },
+  header: {
+    color: 'black'
+  },
+  button: {
+    margin: 12
+  }
+};
+
+/**
+* Decoupled method responsible for generating the list of Artworks per section.
+* @param groupedArtworks, groupBy, size, onExpand
+* @return list of ArtworkListItem separated by Subheader (sections).
+**/
+function generateList(groupedArtworks, groupBy, size, onExpand) {
+  const itemList = [];
```

Comment:
The current setup is paginated, but the user can choose to show all Items by pressing the "Show All" button, that would render all 1000's of Artworks. A way of solving it is using something like react-virtualized (https://github.com/bvaughn/react-virtualized) that would just render what is being shown. If using the library is not possible, this could be implemented by listening to the scroll events and rendering new items when needed.